exporter/otlp/lib/opentelemetry/exporter/otlp/exporter.rb

```diff
@@ -47,22 +47,16 @@ def self.ssl_verify_mode
           end
         end
 
-        def initialize(endpoint: OpenTelemetry::Common::Utilities.config_opt('OTEL_EXPORTER_OTLP_TRACES_ENDPOINT', 'OTEL_EXPORTER_OTLP_ENDPOINT', default: 'http://localhost:4318/v1/traces'),
+        def initialize(endpoint: nil,
                        certificate_file: OpenTelemetry::Common::Utilities.config_opt('OTEL_EXPORTER_OTLP_TRACES_CERTIFICATE', 'OTEL_EXPORTER_OTLP_CERTIFICATE'),
                        ssl_verify_mode: Exporter.ssl_verify_mode,
                        headers: OpenTelemetry::Common::Utilities.config_opt('OTEL_EXPORTER_OTLP_TRACES_HEADERS', 'OTEL_EXPORTER_OTLP_HEADERS', default: {}),
                        compression: OpenTelemetry::Common::Utilities.config_opt('OTEL_EXPORTER_OTLP_TRACES_COMPRESSION', 'OTEL_EXPORTER_OTLP_COMPRESSION', default: 'gzip'),
                        timeout: OpenTelemetry::Common::Utilities.config_opt('OTEL_EXPORTER_OTLP_TRACES_TIMEOUT', 'OTEL_EXPORTER_OTLP_TIMEOUT', default: 10),
                        metrics_reporter: nil)
-          raise ArgumentError, "invalid url for OTLP::Exporter #{endpoint}" unless OpenTelemetry::Common::Utilities.valid_url?(endpoint)
+          @uri = determine_url(endpoint)
```

Comment:
```suggestion
          endpoint ||= ENV['OTEL_EXPORTER_OTLP_TRACES_ENDPOINT']
          @uri = begin
                   if endpoint.nil?
                     endpoint = ENV['OTEL_EXPORTER_OTLP_ENDPOINT'] || 'http://localhost:4318'
                     URI.join(endpoint, 'v1/traces')
                   elsif endpoint.strip.empty?
                     raise ArgumentError, "invalid url for OTLP::Exporter #{endpoint}"
                   else
                     URI(endpoint)
                   end
                 rescue URI::Error
                   raise ArgumentError, "invalid url for OTLP::Exporter #{endpoint}"
                 end
```
I think this captures the essence of the rules while raising as close to the public interface as possible.

Reply:
Yes, that's better (using some language features that I didn't know yet, nice :smile:). Thanks for the suggestion :raised_hands: 

Adapted the suggesting in [40204f9](https://github.com/open-telemetry/opentelemetry-ruby/pull/1506/commits/40204f970c45710608f2f8fbae95d0fbe4e557dc), but included appending `/` (see https://github.com/open-telemetry/opentelemetry-ruby/pull/1506#discussion_r1289966425)